bson/primitive/decimal.go

```diff
@@ -114,6 +121,80 @@ Loop:
 	return string(repr[last+pos:])
 }
 
+// BigInt returns significand as big.Int and exponent, bi * 10 ^ exp.
+func (d Decimal128) BigInt() (bi *big.Int, exp int, err error) {
+	h, l := d.GetBytes()
+	var pos int // positive sign
+
+	if h>>63&1 == 0 {
+		pos = 1
+	}
+
+	switch h >> 58 & (1<<5 - 1) {
+	case 0x1F:
+		return nil, 0, fmt.Errorf("cannot parse NaN as a *big.Int")
+	case 0x1E:
+		return nil, 0, fmt.Errorf("cannot parse %s as a *big.Int", "-Infinity"[pos:])
```

Comment:
This line is kind of clever, but I'd prefer something more direct and more obvious, since this had me pause for a bit to realize what was going on with the `[pos:]`.

Specifically, I'd probably prefer to create two strings for the two infinities and conditionally put them into the error depending on `pos`.